oom notification system docs (#3664)
kyma-project/test-infra: development/pubsub-connector/pubsubConnector/README.md

# Overview

> This chart lets you install the PubSub Connector on [Kyma](https://kyma-project.io/). PubSub Connector is pulling messages from PubSub subscription through pubsubGateway. It translate messages to CloudEvents compliant events and publish them to Kyma Event Publisher Proxy. These events are consumed by Kyma serverless [*oomFoundFunction*](../oomFoundFunction). OomFound function will extracts data from event and send notification to slack channel. To do so, function is utilising Slack Connector Application APIs.

Reviewer: ```suggestion
This chart lets you install the PubSub Connector on [Kyma](https://kyma-project.io/). The PubSub Connector pulls messages from a PubSub subscription through the PubSub Gateway. It translates the messages to Cloud Events and publishes them to the Kyma Event Publisher Proxy. These events are consumed by the Kyma Serverless [*oomFoundFunction*](../oomFoundFunction), which extracts data from the event and sends a notification to a Slack channel. The Function utilizes the Slack Connector Application APIs to send notifications.
```

## Prerequisites

1. PubSub Connector utilizes Kyma features. You need a running Kyma cluster. See the Kyma documentation to learn how to [install](https://kyma-project.io/docs/#installation-installation) it.

> 2. It uses PubSub Gateway for pulling messages from PubSub subscription. See PubSub Gateway [installation](../pubSubGateway/README.md) to learn how to build it.

Reviewer: ```suggestion
2. The PubSub Connector uses the PubSub Gateway to pull messages from a PubSub subscription. See the [PubSub Gateway installation](../pubSubGateway/README.md) document to learn how to build it.
```

> 3. [Helm3](https://helm.sh/docs/intro/install/) installed on your workstation.

Reviewer: ```suggestion
3. You need to have [Helm3](https://helm.sh/docs/intro/install/) installed on your workstation.
```

> 4. The Connector publishes messages on a Slack channel. You need a Slack app with the `chat:write` OAuth scope assigned to bot token. See the [Slack documentation](https://api.slack.com/authentication/basics) to learn how to create a new app.

Reviewer: ```suggestion
4. The Connector publishes messages on a Slack channel. You need a Slack app with the `chat:write` OAuth scope assigned to a bot token. See the [Slack documentation](https://api.slack.com/authentication/basics) to learn how to create a new app.
```

## Installation

1. Clone the `kyma-project/test-infra` repository by running `git clone git@github.com:kyma-project/test-infra.git`.

> 2. Create [Applications](https://kyma-project.io/docs/components/application-connector/#tutorials-create-a-new-application) resources.

Reviewer: ```suggestion
2. Create [Application](https://kyma-project.io/docs/components/application-connector/#tutorials-create-a-new-application) resources.
```

> 3. Define Slack API [service](https://kyma-project.io/docs/components/application-connector/#tutorials-register-a-service-register-an-api-with-a-specification-url).

Reviewer: ```suggestion
3. Define a Slack [API service](https://kyma-project.io/docs/components/application-connector/#tutorials-register-a-service-register-an-api-with-a-specification-url).
```

4. Run `cd test-infra/development/pubsub-connector`.

> 5. Create a namespace: `kubectl create ns gh-connector`

Reviewer: ```suggestion
5. Create a Namespace by running `kubectl create ns gh-connector`
```

> 6. Configure Kyma to use [external](https://kyma-project.io/docs/components/serverless/#tutorials-set-an-external-docker-registry) image repository.

Reviewer: ```suggestion
6. Configure Kyma to use an [external image repository](https://kyma-project.io/docs/components/serverless/#tutorials-set-an-external-docker-registry).
```
